dps310/dps310.go

```diff
@@ -0,0 +1,539 @@
+	}
+
+	// continuous mode
+	err = d.SetMode(ContPresTempMeas)
+	if err != nil {
+		return
+	}
+
+	d.waitDataAvailable(10 * time.Millisecond)
+
+	// set sealevelPressure
+	d.seaLevelPressure = 1013.25
+
+	return
+}
+
+// waits for pressure and temperature data available
+func (d *DPS310) waitDataAvailable(waitTime time.Duration) (err error) {
+	var (
+		tOk, pOk bool
+	)
+	for {
+		// wait until we have at least one good measurement
+		tOk, err = d.TemperatureAvailable()
+		if err != nil {
+			return
+		}
+		pOk, err = d.PressureAvailable()
+		if err != nil {
+			return
+		}
+		if tOk && pOk {
+			break
+		}
+		time.Sleep(waitTime)
+	}
+	return nil
+}
+
+func (d *DPS310) SetMode(mode Mode) error {
+
+	// Check if mode is possible
+	if mode == Invalid1 || mode == Invalid2 {
+		return fmt.Errorf("invalid measuring mode: %v", mode)
+	}
+	if d.pressureOs+d.tempOs > 0 && mode < ContPresMeas {
+		return fmt.Errorf("incompatible measuring mode: %v for sample rates > 1", mode)
+	}
+
+	data, err := d.readRegister(MEAS_CFG)
+	if err != nil {
+		return err
+	}
+
+	data &= MEAS_CFG_CTRL_CLR_MSK
+	mode &=  MEAS_CFG_CTRL_SET_MSK
+	data = data | uint8(mode)
+	return d.writeRegister(MEAS_CFG, data)
+}
+
+// readRegister reads from a single  register.
+func (d *DPS310) readRegister(address uint8) (uint8, error) {
+	buf := d.rwBuf[:2]
+	buf[0] = SPIReadBit| address
+	buf[1] = 0
+	err := d.tx(buf)
+	if err != nil {
+		return 0x0, err
+	}
+	return buf[1], nil
+}
+
+// writeRegister writes a single byte to a register.
+func (d *DPS310) writeRegister(address, data uint8) (err error) {
+	buf := d.rwBuf[:2]
+	buf[0] = address
+	buf[1] = data
+	return d.tx(buf)
+}
+
+func (d *DPS310) tx(buf []uint8) (err error) {
+	d.cs.Low()
+	time.Sleep(CSB_Setup_Time * time.Nanosecond)
+
+	err = d.spi.Tx(buf, buf)
+
+	time.Sleep(CSB_Setup_Time * time.Nanosecond)
+	d.cs.High()
+	return
+}
+
+func (d *DPS310) Reset() (err error) {
+
+	// check func for sesnor ready
+	sensorRdy :=
+		func() (bool, error) {
+			v, err := d.readRegister(MEAS_CFG)
+			if err != nil {
+				return false, err
+			}
+			return v & MEAS_CFG_SENSOR_RDY_MSK != 0, nil
+		}
+
+	// err = d.writeRegister(RESET, 0b1000_1001)
+	err = d.writeRegister(RESET, RESET_VAL)
+	if err != nil {
+		return
+	}
+
+	// Wait for a bit after reset
+	time.Sleep(10 * time.Millisecond)
+
+	// Wait for Sensor getting ready
+	rdy := false
+	for {
+		rdy, err = sensorRdy()
+		if err != nil {
+			return
+		}
+		if rdy {
+			break
+		}
+		time.Sleep(20 * time.Millisecond)
+	}
+	return
+}
+
+func (d *DPS310) readCalibration() (err error) {
+
+	// check func for calibration ready
+	calibRdy := func() (bool, error) {
+		v, err := d.readRegister(MEAS_CFG)
+		if err != nil {
+			return false, err
+		}
+		return v & MEAS_CFG_COEF_RDY_MSK != 0, nil
+	}
+
+	// Wait till we're ready to read calibration
+	rdy := false
+	for {
+		rdy, err = calibRdy()
+		if err != nil {
+			return
+		}
+		if rdy {
+			break
+		}
+		time.Sleep(20 * time.Millisecond)
```

Comment:
maybe add a comment here explaining that the max time according to the datasheet is `40ms`